sihl_contrib/session/lib/repo/mariadb/repo.ml

```diff
@@ -1 +1,134 @@
-(* TODO *)
+module Sql = struct
+  (* TODO move to some common mariadb namespace *)
+  let set_fk_check connection status =
+    let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+    let request =
+      Caqti_request.exec Caqti_type.bool
+        {sql|
+        SET FOREIGN_KEY_CHECKS = ?;
+           |sql}
+    in
+    Connection.exec request status
+
+  module Session = struct
+    module Model = struct
+      open Sihl_session.Model.Session
+
+      let t =
+        let encode m = Ok (m.key, m.data, m.expire_date) in
+        let decode (key, data, expire_date) = Ok { key; data; expire_date } in
+        Caqti_type.(custom ~encode ~decode (tup3 string string pdate))
+    end
+
+    let get_all connection =
+      let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+      let request =
+        Caqti_request.find Caqti_type.unit Model.t
+          {sql|
+        SELECT
+          session_key,
+          session_data,
+          expire_date
+        FROM sessions_sessions
+        |sql}
+      in
+      Connection.collect_list request
+
+    let get connection =
+      let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+      let request =
+        Caqti_request.find_opt Caqti_type.string Model.t
+          {sql|
+        SELECT
+          session_key,
+          session_data,
+          expire_date
+        FROM sessions_sessions
+        WHERE sessions_sessions.key = ?
+        |sql}
+      in
+      Connection.find_opt request
+
+    let upsert connection =
+      let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+      let request =
+        Caqti_request.exec Model.t
+          {sql|
+        INSERT INTO sessions_sessions (
+          session_key,
+          session_data,
+          expire_date
+        ) VALUES (
+          ?,
+          ?,
+          ?
+        ) ON CONFLICT (key) DO UPDATE SET
+        session_data = ?
+        |sql}
+      in
+      Connection.exec request
+
+    let delete connection =
+      let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+      let request =
+        Caqti_request.exec Caqti_type.string
+          {sql|
+      DELETE FROM sessions_sessions
+      WHERE sessions_sessions.key = ?
+      |sql}
+      in
+      Connection.exec request
+
+    let clean connection =
+      let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+      let request =
+        Caqti_request.exec Caqti_type.unit
+          {sql|
+TRUNCATE sessions_session;
+|sql}
```

Comment:
Wrong indentation? Or maybe earlier indentation for SQL blocks are wrong?

Reply:
Fixed.